src/wkok/openai_clojure/azure.clj

```diff
@@ -0,0 +1,42 @@
+(ns wkok.openai-clojure.azure
+  (:require
+   [cheshire.core :as json]
+   [clojure.java.io :as io]
+   [clojure.set :as set]
+   [martian.clj-http :as martian-http]
+   [martian.core :as martian]))
+
+(def add-authentication-header
+  {:name ::add-authentication-header
+   :enter (fn [ctx]
+            (assoc-in ctx [:request :headers "api-key"]
+                      (System/getenv "AZURE_OPENAI_API_KEY")))})
+
+
+
+(defn patch-handler [m]
+  (let [completion-handler (-> m :handlers first)
```

Comment:
completion-handler might not always be first, maybe consider using `some` or `filter` instead?

Reply:
fixed